dateglob/__init__.py

```diff
@@ -205,19 +210,21 @@ def extract_full_years(dates):
         year_to_dates[d.year].add(d)
 
     full_years = set()
```

Comment:
I'm surprised `full_years` doesn't need a type.

Reply:
Yeah, turns out it can be inferred :)